### PR TITLE
user, push, pull event extensions

### PR DESCRIPTION
**Pull request by seregamorph**
# Description 
* GitHub event `pull_request` (`GHEventPayload.PullRequest`) new field `label` of type GHLabel (may present when `"action":"labeled"`); example json with test provided
* GitHub event `push` (`GHEventPayload.Push`) new field `compare`
* `GitUser` new field `username` (e.g. push.commits[*].author.username)

# Before submitting a PR:
We love getting PRs, but we hate asking people for the same basic changes every time. 

- [ ] Push your changes to a branch other than `master`. Create your PR from that branch.    
- [ ] Add JavaDocs and other comments
- [ ] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [ ] Run `mvn clean compile` locally. This may reformat your code, commit those changes.
- [ ] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.

# When creating a PR: 

- [ ] Fill in the "Description" above. 
- [ ] Enable "Allow edits from maintainers".

### CODE REVIEW

**Comment by bitwiseman** [comment]
@seregamorph 
Maybe you could split out the `changes` change into a separate PR? We could do the obvious modifications first.

File: src/main/java/org/kohsuke/github/GitUser.java [hunk 1]
```diff
@@ -27,14 +27,23 @@ public String getName() {
...
         return email;
     }
 
+    /**
+     * Gets username.
+     *
+     * @return GitHub username
+     */
+    public String getUsername() {
+        return username;
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GitUser.java:44]: I don't see this field in the git user information (https://docs.github.com/en/rest/reference/git#get-a-commit).  Where are you finding this?

Comment (bitwiseman) [reply src/main/java/org/kohsuke/github/GitUser.java:44]: Hm, I see it here: https://github.com/hub4j/github-api/blob/19e1a00fd7c522f94161b153de53accc15b62dfc/src/test/resources/org/kohsuke/github/GHEventPayloadTest/push.json#L21 but it isn't in the current GitHub docs.  Might be only present in older versions of GitHub Server?

File: src/main/java/org/kohsuke/github/GHEventPayload.java [hunk 2]
```diff
@@ -1387,6 +1409,15 @@ public void setPusher(Pusher pusher) {
             this.pusher = pusher;
         }
 
+        /**
+         * Gets compare.
+         *
+         * @return compare
+         */
+        public String getCompare() {
+            return compare;
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHEventPayload.java:1418]: https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#push
Has `compare` in the example but not in the documented fields.  Not sure which is in error. 😢

**Comment by seregamorph** [comment]
`changes` related part was erased (force pushed) from the scope

File: src/main/java/org/kohsuke/github/GitUser.java [hunk 3]
```diff
@@ -15,10 +15,10 @@
 @SuppressFBWarnings(value = { "UWF_UNWRITTEN_PUBLIC_OR_PROTECTED_FIELD", "UWF_UNWRITTEN_FIELD", "NP_UNWRITTEN_FIELD" },
         justification = "JSON API")
 public class GitUser {
-    private String name, email, date;
+    private String name, email, date, username;
 
     /**
-     * Gets name.
+     * Gets name (e.g. from git commit).
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GitUser.java:21]: The commit aspect is mentioned on line 8 above.  I see you're trying to be clear.  Maybe the suggestion below?

```suggestion
     * Gets the git user name for an author or committer on a git commit.
```

File: src/main/java/org/kohsuke/github/GitUser.java (hunk 1 above)
Comment (seregamorph) [reply src/main/java/org/kohsuke/github/GitUser.java:44]: I have it in GitHub (global; not enterprise) standard webhook notification on `push` event

File: src/main/java/org/kohsuke/github/GitUser.java [hunk 4]
```diff
@@ -27,14 +27,23 @@ public String getName() {
     }
 
     /**
-     * Gets email.
+     * Gets email (e.g. from git commit).
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GitUser.java:30]:
```suggestion
     * Gets the git email for an author or committer on a git commit.
```

**Review by bitwiseman — CHANGES_REQUESTED** [review]
Still a few questions left to resolve. Thanks for working on this.

File: src/main/java/org/kohsuke/github/GHEventPayload.java (hunk 2 above)
Comment (seregamorph) [reply src/main/java/org/kohsuke/github/GHEventPayload.java:1407]: You can check the `Webhook payload example` on the link that you gave, the field is here (but not listed in description, right)

File: src/main/java/org/kohsuke/github/GitUser.java (hunk 4 above)
Comment (seregamorph) [reply src/main/java/org/kohsuke/github/GitUser.java]: done

File: src/main/java/org/kohsuke/github/GitUser.java (hunk 1 above)
Comment (bitwiseman) [reply src/main/java/org/kohsuke/github/GitUser.java:44]: @seregamorph 
Interesting. Is it possible it is only present on events?  
This doesn't include it: https://api.github.com/repos/octocat/Hello-World/git/commits/7fd1a60b01f91b314f59955a4e4d4e80d8edf11d

I'd be okay with adding this if the javadoc clearly states that is only seen on events and we add `@Preview @Deprecated` annotations to it.  I would also ask that you contact GitHub support and ask them to clarify the documentation. 


username

**Comment by bitwiseman** [comment]
@seregamorph 
Did see this?  https://github.com/hub4j/github-api/pull/944/files#r494672997 
You can change it, we can discuss further in the PR, or you can split that part of the change into another PR and we can continue discussion there.  Up to you.  Thanks for contributing!

**Comment by seregamorph** [comment]
> @seregamorph
> Did see this? https://github.com/hub4j/github-api/pull/944/files#r494672997
> You can change it, we can discuss further in the PR, or you can split that part of the change into another PR and we can continue discussion there. Up to you. Thanks for contributing!

Yes. I raised a question in https://github.community, but it still waits for moderation.
Screenshot 2020-09-26 at 11 55 15

So let's wait a bit.

File: src/main/java/org/kohsuke/github/GitUser.java (hunk 1 above)
Comment (seregamorph) [reply src/main/java/org/kohsuke/github/GitUser.java:49]: Done. Marked as
```
    @Preview
    @Deprecated
    @CheckForNull
```
and added javadoc

**Comment by seregamorph** [comment]
GitHub community topic is still under review and is not published (already 3 days). Anything else I can do here now?

File: src/main/java/org/kohsuke/github/GitUser.java (hunk 1 above)
Comment (schernov-xo) [reply src/main/java/org/kohsuke/github/GitUser.java:49]: The topic is published https://github.community/t/push-event-structure-author-committer-id-for-commits/134109 but there is still no answers.

**Review by bitwiseman — APPROVED** [review]

**Pull request merged**